reporter.go

```diff
@@ -38,3 +39,20 @@ func NewRequireReporter(t require.TestingT) *RequireReporter {
 func (r *RequireReporter) Errorf(message string, args ...interface{}) {
 	r.backend.FailNow(fmt.Sprintf(message, args...))
 }
+
+// FatalReporter is a struct that implements the Reporter interface
+// and calls t.Fatalf() when a test fails.
+type FatalReporter struct {
+	backend testing.TB
+}
+
+// NewFatalReporter returns a new FatalReporter object.
+func NewFatalReporter(backend testing.TB) *FatalReporter {
+	return &FatalReporter{backend}
+}
+
+// Errorf implements Reporter.Errorf .
+func (r *FatalReporter) Errorf(message string, args ...interface{}) {
+	r.backend.Logf(message, args...)
+	r.backend.Fail()
```

Comment:
Can't we simplify this to `r.backend.Fatalf`?

Reply:
I get runtime.Goexit issue with the test if I do so

Reply:
Oh, I see; that's because Fail() just marks test as failed. We instead need Fatalf or FailNow.

So we indeed need to wrap testing.T and override Fatalf, as I suggested initially.

Reply:
oK I will overwrite it in  the test file right?

Reply:
also what will look like the overwrited method